backend/app/routers/health.py

```
@@ -25,17 +25,21 @@ def check_sources():
+        from app.services.alpha_vantage_client import get_alpha_vantage_api_key
+        api_key = get_alpha_vantage_api_key()
+        if not api_key:
+            return {"available": False, "error": "API key not configured"}
         resp = _requests.get(
-            "https://query1.finance.yahoo.com/v8/finance/chart/AAPL?range=1d&interval=1d",
-            headers={"User-Agent": "Mozilla/5.0"},
+            "https://www.alphavantage.co/query",
+            params={"function": "EARNINGS_CALENDAR", "symbol": "AAPL", "horizon": "3month", "apikey": api_key},
             timeout=10,
         )
         return {"available": resp.status_code == 200, "error": None}
```

Ruff (0.15.4)

[warning] Consider moving this statement to an `else` block

(TRY300)